back/lib/tasks/data_migrate/20231127_update_idea_statuses_and_phases.rake

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+# bin/rails "data_migrate:update_idea_statuses_and_phases['ideas.csv','participatie.leuven.be','nl-BE']"
+namespace :data_migrate do
+  desc 'Update Idea statuses and phases.'
+  task :update_idea_statuses_and_phases, %i[url host locale] => [:environment] do |_t, args|
+    data = CSV.parse(open(args[:url]).read, { headers: true, col_sep: ',', converters: [] })
+
+    Apartment::Tenant.switch(args[:host].tr('.', '_')) do
+      default_locale = args[:locale]
+      statuses = IdeaStatus.all.to_a
+      find_status_by_title = ->(title) { statuses.find { _1.title_multiloc[default_locale] == title } || raise("Idea status not found: '#{title}'.") }
+      create_status_by_title = lambda do |title, color|
+        IdeaStatus.find_or_create_by!(
+          title_multiloc: { default_locale => title },
+          description_multiloc: { default_locale => title },
+          color: color,
+          code: 'custom'
+        )
+      end
+
+      statuses_mapping = {
+        'verder onderzocht' => find_status_by_title.call('Wordt Onderzocht'),
+        'uitgevoerd' => find_status_by_title.call('uitgevoerd'),
+        'afgesloten/wordt niet verder onderzocht' => find_status_by_title.call('Afgesloten'),
+        'ingepland' => find_status_by_title.call('Ingepland'),
+        'uitvoering in 2024' => create_status_by_title.call('Uitvoering in 2023', '#216db7'),
+        'uitvoering in 2023' => create_status_by_title.call('Uitvoering in 2024', '#b721a7')
+      }
```

Comment:
Nothing per se wrong with this, but just sharing another approach: I would have done the idea status mapping in the spreadsheet file, to not risk creating 100 new statuses in case there is a mistake, and getting immediate feedback.

So I mean, adding a status_id column and then just mapping with a formula from the text to the status ID.

This works too, just a bit more error prone.

Reply:
Actually, I ran it yesterday already, because we agreed it was urgent. Gladly, there are only 2 new statuses :D https://participatie.leuven.be/nl-BE/admin/settings/statuses